From 325b306f4279e0b85441d0d6314b99edd0c7d0fd Mon Sep 17 00:00:00 2001
From: Pierre-Henry Moussay <ph.moussay.dev@gmail.com>
Date: Wed, 14 Feb 2024 12:58:10 +0000
Subject: [PATCH 2/2] riscv: dts: microchip: add mpfs-disco-kit dts

The MPFS-DISCO-KIT (PolarFire SoC Discovery Kit) is a compact SoC
prototyping board featuring a Microchip PolarFire SoC
MPFS095T-1FCSG325E. Features include:
- 1 GB DDR4 SDRAM
- Gigabit Ethernet
- microSD-card slot

Signed-off-by: Pierre-Henry Moussay <ph.moussay.dev@gmail.com>
---
 arch/riscv/boot/dts/microchip/Makefile        |   1 +
 .../dts/microchip/mpfs-disco-kit-fabric.dtsi  |  17 ++
 .../boot/dts/microchip/mpfs-disco-kit.dts     | 170 ++++++++++++++++++
 3 files changed, 188 insertions(+)
 create mode 100644 arch/riscv/boot/dts/microchip/mpfs-disco-kit-fabric.dtsi
 create mode 100644 arch/riscv/boot/dts/microchip/mpfs-disco-kit.dts

diff --git a/arch/riscv/boot/dts/microchip/Makefile b/arch/riscv/boot/dts/microchip/Makefile
index 6ab195c1735f..0ad0fda286cd 100644
--- a/arch/riscv/boot/dts/microchip/Makefile
+++ b/arch/riscv/boot/dts/microchip/Makefile
@@ -4,6 +4,7 @@ dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-icicle-kit.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-icicle-kit-context-a.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-m100pfsevp-sdcard.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-m100pfsevp-emmc.dtb
+dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-disco-kit.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-polarberry.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-video-kit.dtb
 dtb-$(CONFIG_SOC_MICROCHIP_POLARFIRE) += mpfs-tysom-m.dtb
diff --git a/arch/riscv/boot/dts/microchip/mpfs-disco-kit-fabric.dtsi b/arch/riscv/boot/dts/microchip/mpfs-disco-kit-fabric.dtsi
new file mode 100644
index 000000000000..6685f511b988
--- /dev/null
+++ b/arch/riscv/boot/dts/microchip/mpfs-disco-kit-fabric.dtsi
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/* Copyright (c) 2020-2021 Microchip Technology Inc */
+
+/ {
+      compatible = "microchip,mpfs-disco-kit", "microchip,mpfs";
+
+      fabric-bus@40000000 {
+              compatible = "simple-bus";
+              #address-cells = <2>;
+              #size-cells = <2>;
+              ranges = <0x0 0x40000000 0x0 0x40000000 0x0 0x20000000>, /* FIC3-FAB */
+                       <0x0 0x60000000 0x0 0x60000000 0x0 0x20000000>, /* FIC0, LO */
+                       <0x0 0xe0000000 0x0 0xe0000000 0x0 0x20000000>, /* FIC1, LO */
+                       <0x20 0x0 0x20 0x0 0x10 0x0>, /* FIC0,HI */
+                       <0x30 0x0 0x30 0x0 0x10 0x0>; /* FIC1,HI */
+      };
+};
diff --git a/arch/riscv/boot/dts/microchip/mpfs-disco-kit.dts b/arch/riscv/boot/dts/microchip/mpfs-disco-kit.dts
new file mode 100644
index 000000000000..48e065cd7165
--- /dev/null
+++ b/arch/riscv/boot/dts/microchip/mpfs-disco-kit.dts
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/* Copyright (c) 2020-2021 Microchip Technology Inc */
+
+/dts-v1/;
+
+#include "mpfs.dtsi"
+#include "mpfs-disco-kit-fabric.dtsi"
+
+/* Clock frequency (in Hz) of the rtcclk */
+#define RTCCLK_FREQ           1000000
+
+/ {
+      #address-cells = <2>;
+      #size-cells = <2>;
+      model = "Microchip PolarFire-SoC Discovery Kit";
+      compatible = "microchip,mpfs-disco-kit", "microchip,mpfs";
+
+      soc {
+              dma-ranges = <0x14 0x0 0x0 0x80000000 0x0 0x4000000>,
+                           <0x14 0x4000000 0x0 0xc4000000 0x0 0x6000000>,
+                           <0x14 0xa000000 0x0 0x8a000000 0x0 0x8000000>,
+                           <0x14 0x12000000 0x14 0x12000000 0x0 0x10000000>,
+                           <0x14 0x22000000 0x10 0x22000000 0x0 0x1e000000>;
+      };
+
+      aliases {
+              ethernet0 = &mac0;
+              serial4 = &mmuart4;
+      };
+
+      chosen {
+              stdout-path = "serial4:115200n8";
+      };
+
+      cpus {
+              timebase-frequency = <RTCCLK_FREQ>;
+      };
+
+      kernel: memory@80000000 {
+              device_type = "memory";
+              reg = <0x0 0x80000000 0x0 0x4000000>;
+      };
+
+      ddr_cached_low: memory@8a000000 {
+              device_type = "memory";
+              reg = <0x0 0x8a000000 0x0 0x8000000>;
+      };
+
+      ddr_non_cached_low: memory@c4000000 {
+              device_type = "memory";
+              reg = <0x0 0xc4000000 0x0 0x6000000>;
+      };
+
+      ddr_cached_high: memory@1022000000 {
+              device_type = "memory";
+              reg = <0x10 0x22000000 0x0 0x1e000000>;
+      };
+
+      ddr_non_cached_high: memory@1412000000 {
+              device_type = "memory";
+              reg = <0x14 0x12000000 0x0 0x10000000>;
+      };
+
+      reserved-memory {
+              #address-cells = <2>;
+              #size-cells = <2>;
+              ranges;
+
+              hss: hss-buffer@103fc00000 {
+                      compatible = "shared-dma-pool";
+                      reg = <0x10 0x3fc00000 0x0 0x400000>;
+                      no-map;
+              };
+
+              dma_non_cached_low: non-cached-low-buffer {
+                      compatible = "shared-dma-pool";
+                      size = <0x0 0x4000000>;
+                      no-map;
+                      linux,dma-default;
+                      alloc-ranges = <0x0 0xc4000000 0x0 0x4000000>;
+              };
+
+              dma_non_cached_high: non-cached-high-buffer {
+                      compatible = "shared-dma-pool";
+                      size = <0x0 0x10000000>;
+                      no-map;
+                      linux,dma-default;
+                      alloc-ranges = <0x14 0x12000000 0x0 0x10000000>;
+              };
+
+              fabricbuf0ddrc: buffer@88000000 {
+                      compatible = "shared-dma-pool";
+                      reg = <0x0 0x88000000 0x0 0x2000000>;
+                      no-map;
+              };
+
+              fabricbuf1ddrnc: buffer@c8000000 {
+                      compatible = "shared-dma-pool";
+                      reg = <0x0 0xc8000000 0x0 0x2000000>;
+                      no-map;
+              };
+
+              fabricbuf2ddrncwcb: buffer@d8000000 {
+                      compatible = "shared-dma-pool";
+                      reg = <0x0 0xd8000000 0x0 0x2000000>;
+                      no-map;
+              };
+      };
+
+      udmabuf0 {
+              compatible = "ikwzm,u-dma-buf";
+              device-name = "udmabuf-ddr-c0";
+              minor-number = <0>;
+              size = <0x0 0x2000000>;
+              memory-region = <&fabricbuf0ddrc>;
+              sync-mode = <3>;
+      };
+
+      udmabuf1 {
+              compatible = "ikwzm,u-dma-buf";
+              device-name = "udmabuf-ddr-nc0";
+              minor-number = <1>;
+              size = <0x0 0x2000000>;
+              memory-region = <&fabricbuf1ddrnc>;
+              sync-mode = <3>;
+      };
+
+      udmabuf2 {
+              compatible = "ikwzm,u-dma-buf";
+              device-name = "udmabuf-ddr-nc-wcb0";
+              minor-number = <2>;
+              size = <0x0 0x2000000>;
+              memory-region = <&fabricbuf2ddrncwcb>;
+              sync-mode = <3>;
+      };
+};
+
+&mbox {
+      status = "okay";
+};
+
+&mmc {
+      dma-noncoherent;
+      bus-width = <4>;
+      disable-wp;
+      cap-sd-highspeed;
+      cap-mmc-highspeed;
+      sd-uhs-sdr12;
+      sd-uhs-sdr25;
+      sd-uhs-sdr50;
+      sd-uhs-sdr104;
+      no-1-8-v;
+      status = "okay";
+};
+
+&mmuart4 {
+      status = "okay";
+};
+
+&refclk {
+      clock-frequency = <125000000>;
+};
+
+&rtc {
+      status = "okay";
+};
+
+&syscontroller {
+      status = "okay";
+};
--
2.30.2

